gcs/innards/state.hh

```diff
@@ -535,8 +535,36 @@ namespace gcs::innards
 
         ///@}
 
+        /**
+         * \name Constraint state related functions.
+         */
+        ///@{
+
+        /**
+         * Store a given std::any value as a constraint state that is accessible via
+         * the returned handle and restores on backtrack.
+         */
         [[nodiscard]] auto add_constraint_state(const ConstraintState c) -> ConstraintStateHandle;
+
+        /**
+         * Store a given std::any value as a constraint state that is accessible via
+         * the returned handle and does not restore on backtrack.
+         */
+        [[nodiscard]] auto add_persistent_constraint_state(const ConstraintState c) -> ConstraintStateHandle;
```

Comment:
What exactly is the need for this? In other constraints I've been moving the appropriate data structure into the lambda, so it's stored there instead. Does this get too ugly?

Reply:
Was trying to simplify the arguments being passed around and keep the persistent and dynamic data together as handles until they are needed. Maybe not the best but would require quite a bit of refactoring to change